ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,95 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+import six
+import abc
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
+
+    def configure(self, sssd, mkhomedir, statestore):
+        if not sssd:
+            logger.error("no-sssd options is not supported by installer. "
+                         "For client installation without sssd check the "
+                         "output of 'ipa-advise config-fedora-authconfig'")
+            return
+            # raise ScriptError("Non-sssd installation is not supported")
+
+        # Do not raise exception since authselect may fail just
+        # because there is PAM configuration already in place.
+        if mkhomedir:
+            ipautil.run(
+                [
+                    paths.AUTHSELECT,
+                    "select", "sssd",
+                    "with-mkhomedir",
+                    "--force"
+                ]
+            )
+        else:
+            ipautil.run(
+                [
+                    paths.AUTHSELECT,
+                    "select", "sssd",
+                    "--force"
+                ]
+            )
+
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
```

Comment:
If unconfigure does nothing, it's likely that uninstall won't restore the previous state.